src/main/kotlin/uk/co/poolefoundries/baldinggate/core/CoreComponents.kt

```diff
@@ -72,6 +73,19 @@ data class PositionComponent(val x: Int, val y: Int) : Component {
 
 }
 
+data class AnimationComponent(val entity: Entity, var positions: List<PositionComponent> = listOf<PositionComponent>(), var progress: Float = 0F, var isFinished: Boolean = false) : Component {
+    val animationDuration = 0.1F
+
```

Comment:
function can have index exceeds list dimensions problem here if the frame takes > 0.1s to draw so can handle that by this

fun animationStep(delta: Float) {
        this.progress += delta / animationDuration
        val positionIndex = round(this.progress).toInt()
        if (positionIndex >= this.positions.size-1){
            this.isFinished  = true
            entity.add(positions.last())
        }else{
            entity.add(positions[positionIndex])
        }
    }

Reply:
Use tripple backtick to start code blocks in comments:
```kotlin
fun foo() : Int {
    return bar.value()
}
```